### PR TITLE
obs-ffmpeg: Check correct VA-API codec support

### PR DESCRIPTION
**Pull request by kkartaltepe**
### Description

Previously, we just checked for H264 regardless of which codec was selected. This mostly worked, but on Fedora they ship AV1 without H264 or HEVC. In that case users wont see AV1 as an option.

### Motivation and Context

We really should have been checking the right codec, and users will want to use AV1 without special drivers.

### How Has This Been Tested?

I dont see any changes on my AMD machine. I'll test a fedora machine later.

### Types of changes

 - Bug fix (non-breaking change which fixes an issue) 

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c [hunk 1]
```diff
@@ -1074,6 +1074,20 @@ static bool get_device_name_from_pci(struct pci_access *pacc, char *pci_slot,
...
+{
+	switch (codec) {
+	case CODEC_H264:
+		return vaapi_device_h264_supported(path);
+	case CODEC_HEVC:
+		return vaapi_device_hevc_supported(path);
+	case CODEC_AV1:
+		return vaapi_device_av1_supported(path);
+	}
+	return false;
```
Comment (tytan652) [inline plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c:1088]: Nit:

```suggestion
	default:
		return false;
	}
```

Comment (kkartaltepe) [reply plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c:1088]: Actually i want it outside the switch to keep all the branches covered. With default we wont hit the warning if someone adds a new codec (unlikely, but still).

Comment (tytan652) [reply plugins/obs-ffmpeg/obs-ffmpeg-vaapi.c:1088]: Normally we should remove `-Wno-error=switch`, but QSV…

**Review by tytan652 — APPROVED** [review]
LGTM

**Review by Lain-B — APPROVED** [review]

**Pull request merged**